hoodie-spark/src/main/scala/com/uber/hoodie/HoodieSparkSqlWriter.scala

```diff
@@ -52,11 +52,10 @@ private[hoodie] object HoodieSparkSqlWriter {
     if (path.isEmpty || tblName.isEmpty) {
       throw new HoodieException(s"'${HoodieWriteConfig.TABLE_NAME}', 'path' must be set.")
     }
-    val serializer = sparkContext.getConf.get("spark.serializer")
-    if (!serializer.equals("org.apache.spark.serializer.KryoSerializer")) {
-      throw new HoodieException(s"${serializer} serialization is not supported by hoodie. Please use kryo.")
+    sparkContext.getConf.getOption("spark.serializer") match {
+      case Some(ser) if ser.equals("org.apache.spark.serializer.KryoSerializer") =>
+      case _ => throw new HoodieException("hoodie only support org.apache.spark.serializer.KryoSerializer as spark.serializer")
     }
```

Comment:
Does the above code change do the same as the following :  
sparkContext.getConf.getOption("spark.serializer") match {
      case Some(ser) => if ser.equals("org.apache.spark.serializer.KryoSerializer") None 
      case _ => throw new HoodieException("hoodie only support org.apache.spark.serializer.KryoSerializer as spark.serializer")
    }

Reply:
nope

Reply:
Equivalent to
```scala
sparkContext.getConf.getOption("spark.serializer") match {
      case Some(ser) if ser.equals("org.apache.spark.serializer.KryoSerializer") =>
      case Some(ser) if !ser.equals("org.apache.spark.serializer.KryoSerializer") => throw new HoodieException("hoodie only support org.apache.spark.serializer.KryoSerializer as spark.serializer")
      case None => throw new HoodieException("hoodie only support org.apache.spark.serializer.KryoSerializer as spark.serializer")
```

Reply:
your solution need another else to throw exception

Reply:
sounds good